### PR TITLE
Add connector automation support to backend functions

### PR DESCRIPTION
**Pull request by Paveltarno**
> [!NOTE]
> ## Description
>
> Adds Zod schema support for connector (webhook-triggered) automations in the function configuration, including a recursive `trigger_conditions` structure for field-level filtering. The schema uses a forward-compatible operator union so that unknown future operators from the backend still parse without errors. An empty `trigger_conditions` form is rejected (mirroring backend validation), while `null`/`undefined` are accepted to clear conditions.
>
> ## Related Issue
>
> None
>
> ## Type of Change
>
> - [ ] Bug fix (non-breaking change which fixes an issue)
> - [x] New feature (non-breaking change which adds functionality)
> - [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
> - [ ] Documentation update
> - [ ] Refactoring (no functional changes)
> - [ ] Other (please describe):
>
> ## Changes Made
>
> - Added `ConnectorAutomationSchema` to `function/schema.ts` covering `type: "connector"` automations with `integration_type`, `events`, `resource_id`, and `trigger_conditions` fields
> - Added `TriggerConditionSchema` and recursive `TriggerConditionGroupSchema` (via `z.lazy`) for arbitrarily nested and/or condition trees
> - Added `ConditionOperatorSchema` with all known backend operators enumerated plus a forward-compat `z.string()` fallback
> - Included `ConnectorAutomationSchema` in the `AutomationSchema` union alongside existing scheduled and entity types
> - Added `packages/cli/tests/core/function-schema.spec.ts` with 9 tests covering valid, null/undefined, catch-all, recursive, exists-style, and rejection cases
> - Added `.superset/` to `.gitignore`
>
> ## Testing
>
> - [ ] I have tested these changes locally
> - [x] I have added/updated tests as needed
> - [x] All tests pass (`npm test`)
>
> ## Checklist
>
> - [x] My code follows the project's style guidelines
> - [x] I have performed a self-review of my own code
> - [x] I have commented my code, particularly in hard-to-understand areas
> - [ ] I have made corresponding changes to the documentation (if applicable)
> - [x] My changes generate no new warnings
> - [ ] I have updated `docs/` (AGENTS.md) if I made architectural changes
>
> ## Additional Notes
>
> The `events` array intentionally has no `.min(1)` constraint — an empty array means "catch-all" for a connector integration, and per-integration enforcement is handled server-side. The `KnownConditionOperators` list is kept in sync with the backend `ConditionOperator` enum; the `z.string().min(1)` fallback ensures forward compatibility when new operators are added.
>
> ---
> <sub>🤖 Generated by Claude | 2026-04-13 08:06 UTC</sub>

### CODE REVIEW

**Comment by Paveltarno** [comment]
@claude review

**Comment by Paveltarno** [comment]
## Review Response

Thanks for the thorough review! Here's what we addressed and what we intentionally left as-is:

### Addressed

- **`events` has no minimum — no comment explaining why**: Added inline comment: `// No .min(1) — empty means catch-all; the server enforces per-integration.` (9f32013)
- **No negative-case tests**: Added 3 rejection tests — missing `integration_type`, empty conditions group, unknown automation type. We also already had rejection tests for empty trigger condition forms (`{}`, `{ conditions: [] }`) from 24aa8b5. Total test count is now 154. (9f32013)

### Intentionally left as-is

- **`.strict()` on `TriggerConditionSchema`**: We intentionally removed `.strict()` from all schemas in this file for consistency — no other schema (scheduled, entity, base) uses it. The backend validates malformed payloads; the CLI stays lenient to avoid rejecting future API fields on inbound responses (pull/list).
- **TS type allows empty `conditions` but schema enforces `.min(1)`**: The manual `TriggerConditionGroup` type is required for `z.lazy()` recursion — TypeScript can't infer recursive types from Zod schemas. The schema is the runtime source of truth; the type discrepancy is cosmetic and doesn't affect behavior.
- **No exported types for connector schemas**: These are purely internal to the function schema module. No callers need them today — we can export when needed.

**Comment by Paveltarno** [comment]
@claude re-review

File: packages/cli/src/core/resources/function/schema.ts [hunk 1]
```diff
@@ -65,12 +66,75 @@ const EntityAutomationSchema = AutomationBaseSchema.extend({
...
+    }),
+);
+
+// Connector automation (webhook-triggered)
+const ConnectorAutomationSchema = AutomationBaseSchema.extend({
+  type: z.literal("connector"),
+  integration_type: IntegrationTypeSchema,
+  // No .min(1) — empty means catch-all; the server enforces per-integration.
+  events: z.array(z.string()),
+  resource_id: z.string().nullable().optional(),
```
Comment (kfirstri) [inline packages/cli/src/core/resources/function/schema.ts:127]: I don't see anything uses this

File: packages/cli/src/core/resources/function/schema.ts [hunk 2]
```diff
@@ -65,12 +66,75 @@ const EntityAutomationSchema = AutomationBaseSchema.extend({
...
+  trigger_conditions: TriggerConditionGroupSchema.nullable().optional(),
```
Comment (kfirstri) [inline packages/cli/src/core/resources/function/schema.ts:128]: camelCases 🤩

File: packages/cli/src/core/resources/function/schema.ts [hunk 3]
```diff
@@ -65,12 +66,75 @@ const EntityAutomationSchema = AutomationBaseSchema.extend({
...
+  "not_contains",
+  "starts_with",
+  "ends_with",
+  "in_list",
+  "not_in_list",
+  "exists",
+  "not_exists",
+  "is_empty",
+  "is_not_empty",
+] as const;
```
Comment (kfirstri) [inline packages/cli/src/core/resources/function/schema.ts:88]: are all of these supported on the backend? Maybe we should be more lean about this?

**Review by kfirstri — COMMENTED** [review]
looks cool :)

File: packages/cli/src/core/resources/function/schema.ts (hunk 3 above)
Comment (Paveltarno) [reply packages/cli/src/core/resources/function/schema.ts:88]: yeah they are all supported, i've decided to add them after talking with the tech lead responsible on the server as he assured me this data is not bound to change soon, thought it would be a better hint for the LLM but can also drop it, wdyt?

File: packages/cli/src/core/resources/function/schema.ts (hunk 1 above)
Comment (Paveltarno) [reply packages/cli/src/core/resources/function/schema.ts:127]: used in some resources, please see preview docs:
https://github.com/base44-dev/mintlify-docs/pull/1091/changes#diff-6f31d7663328dd924c4188958e9de4d3eb8300c37804c3855097eecbc5ba074bR236

File: packages/cli/src/core/resources/function/schema.ts (hunk 2 above)
Comment (Paveltarno) [reply packages/cli/src/core/resources/function/schema.ts:128]: oh now i remember, it's because the existing backend function schema uses snake_case and didn't want to break.. we're actually in a bit of a mix rn, i suggest we tackle this separately 

```
  ┌─────────────┬────────────────┬──────────────────────────────────────────────────────────────────────────────────────────────┐
  │  Resource   │  Config file   │                                             Case                                             │
  ├─────────────┼────────────────┼──────────────────────────────────────────────────────────────────────────────────────────────┤
  │ Auth Config │ config.jsonc   │ camelCase (enableGoogleLogin, ssoProviderName)                                               │
  ├─────────────┼────────────────┼──────────────────────────────────────────────────────────────────────────────────────────────┤
  │ Connector   │ <type>.jsonc   │ camelCase (type, scopes — single words only though)                                          │
  ├─────────────┼────────────────┼──────────────────────────────────────────────────────────────────────────────────────────────┤
  │ Entity      │ <name>.jsonc   │ camelCase (minLength, maxLength, enumNames) + snake_case in RLS (user_condition, created_by) │
  ├─────────────┼────────────────┼──────────────────────────────────────────────────────────────────────────────────────────────┤
  │ Agent       │ agent.jsonc    │ snake_case (tool_configs, entity_name, whatsapp_greeting)                                    │
  ├─────────────┼────────────────┼──────────────────────────────────────────────────────────────────────────────────────────────┤
  │ Function    │ function.jsonc │ snake_case (function_args, is_active, schedule_mode, integration_type, etc.)                 │
  └─────────────┴────────────────┴──────────────────────────────────────────────────────────────────────────────────────────────┘
```

File: packages/cli/src/core/resources/function/schema.ts (hunk 3 above)
Comment (kfirstri) [reply packages/cli/src/core/resources/function/schema.ts:88]: Lets keep it. I see now in the code that 
```
const ConditionOperatorSchema = z.union([
  z.enum(KnownConditionOperators),
  z.string().min(1),
]);
```

so we're already lean about it with the z.string.min(1) right? it will accept any string?

**Review by kfirstri — APPROVED** [review]
lgtm

File: packages/cli/src/core/resources/function/schema.ts (hunk 3 above)
Comment (Paveltarno) [reply packages/cli/src/core/resources/function/schema.ts:88]: correct (:

Comment (Paveltarno) [reply packages/cli/src/core/resources/function/schema.ts:88]: correct (:
